src/common/pal/CMakeLists.txt

```diff
@@ -0,0 +1,29 @@
+cmake_minimum_required(VERSION 3.12.4)
+project(pal)
+
+if(WIN32)
+    set(OS_NAME "windows")
+elseif(APPLE)
+    set(OS_NAME "macos")
+elseif(UNIX)
+    set(OS_NAME "linux")
+else()
+    message(FATAL_ERROR "Unsupported OS")
+endif()
+
+include_directories(include/${OS_NAME})
```

Comment:
Since this new, we should try `target_include_directories(pal PUBLIC include/${OS_NAME})`

Reply:
Done